geometric_kernels/spaces/grassmannian.py

```diff
@@ -0,0 +1,108 @@
+import lab as B
+import numpy as np
+from opt_einsum import contract as einsum
+
+from geometric_kernels.lab_extras import qr
+from geometric_kernels.spaces.homogeneous_spaces import (
+    CompactHomogeneousSpace,
+    CompactHomogeneousSpaceAddtitionTheorem,
+)
+from geometric_kernels.spaces.so import SOGroup
+
+
+class _SOxSO:
+    """Helper class for sampling, represents SO(n) x SO(m)"""
+
+    def __init__(self, n, m):
+        self.n, self.m = n, m
+        self.so_n = SOGroup(n)
+        self.so_m = SOGroup(m)
+        self.dim = n * (n - 1) // 2 + m * (m - 1) // 2
+
+    def random(self, key, number):
+        key, h_u = self.so_n.random(key, number)
+        key, h_d = self.so_m.random(key, number)
+        zeros = B.zeros(B.dtype(h_u), number, self.n, self.m)
+        zeros_t = B.transpose(zeros)
+        l, r = B.concat(h_u, zeros_t, axis=-2), B.concat(zeros, h_d, axis=-2)
+        res = B.concat(l, r, axis=-1)
+        return key, res
+
+
+class GrassmannianEigenfunctions(CompactHomogeneousSpaceAddtitionTheorem):
+    def _compute_projected_character_value_at_e(self, signature):
+        """
+        Value of character on class of identity element is equal to the dimension of invariant space
+        In case of grassmannian this value always equal to 1, since the space is symmetric.
+        :param signature:
+        :return: int
+        """
+        return 1
+
+
+class Grassmannian(CompactHomogeneousSpace):
+    r"""
+    Class for Grassmannian manifold as SO(n)/(SO(m)xSO(n-m))
+    Elements of manifold represented as nxm matrices, note that
+    X and X x (SO(m) \oplus I_{n-m}) are representatives of the same element
+    """
+
+    def __new__(cls, n, m, key, average_order=1000):
```

Comment:
Do you really ned `__new__`?

Reply:
We do need `__new__` because we are creating a randomized object here. The randomicity is given by `key`, and we need to return the `key` along with the new object